File: app/src/main/java/com/attendance/myproject/begory/adapters/AttendanceAdapter.java
package com.attendance.myproject.begory.adapters;

import android.app.Activity;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.attendance.myproject.begory.R;
import com.attendance.myproject.begory.entites.Attendance;

import java.util.ArrayList;

public class AttendanceAdapter extends RecyclerView.Adapter<AttendanceAdapter.MyViewHolder> {

    private ArrayList<Attendance> mMenuList;
    private Activity mContext;

    public AttendanceAdapter(Activity context, ArrayList<Attendance> menuList) {
        mMenuList = menuList;
        this.mContext = context;
    }

    @Override
    public int getItemCount() {
        return mMenuList.size();
    }

    @Override
    public AttendanceAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent,
                                                             int viewType) {
        // create a new view
        View v = (View) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.row_attendance, parent, false);
        MyViewHolder vh = new MyViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder,final int position) {

        Attendance attendance = mMenuList.get(position);
        holder.mTextView.setText(attendance.getDate());

    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case
        public TextView mTextView;
        public MyViewHolder(View v) {
            super(v);
            mTextView = v.findViewById(R.id.tv_title);
        }
    }
}

File: app/src/main/java/com/attendance/myproject/begory/presentationLayer/main/attendence/AttendanceAdapter.kt
package com.attendance.myproject.begory.presentationLayer.main.attendence

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.attendance.myproject.begory.R
import com.attendance.myproject.begory.R.string.*
import com.attendance.myproject.begory.Utilities.UiManager
import com.attendance.myproject.begory.presentationLayer.main.attendence.takeAttendance.LevelAttendActivity
import com.attendance.myproject.begory.presentationLayer.main.setting.showTakedAttendance.ShowLevelAttendActivity

class AttendanceAdapter(private val mContext: Context, private val mlist: List<Int>)
    : RecyclerView.Adapter<AttendanceAdapter.AttendanceViewHolder?>() {
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): AttendanceViewHolder {
        val view: View = LayoutInflater.from(viewGroup.context)
                .inflate(R.layout.row_menu, viewGroup, false)
        return AttendanceViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: AttendanceViewHolder, i: Int) {
        viewHolder.mTextView?.text = mContext.getString(mlist[i])
    }

    override fun getItemCount(): Int {
        return mlist.size
    }

    inner class AttendanceViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        //public TextView textView;
        //public ImageView imageView;
            var mTextView: TextView? = null
        init {

            mTextView = itemView.findViewById(R.id.tv_title)

            itemView.setOnClickListener(View.OnClickListener(){
                    val pos = adapterPosition;
                    if (pos != RecyclerView.NO_POSITION){
                        val clickedDataItem = mlist[pos];
                        when (mlist[pos]) {
                            a_lev_college -> UiManager.startActivity(mContext, LevelAttendActivity::class.java, mContext.getString(settingType),lev_college)
                            a_lev_Grad -> UiManager.startActivity(mContext,LevelAttendActivity::class.java, mContext.getString(settingType),lev_Grad )

                        }
                        Toast.makeText(it.context, "You clicked $clickedDataItem", Toast.LENGTH_SHORT).show()
                    }
                })

        }
    }}
